pkg/parser/parser_test.go

```diff
@@ -1073,6 +1073,22 @@ AAAAAAAAAAAA5gm5Mg==
 		{"show table t1 partition (p0,p1) distributions", true, "SHOW TABLE `t1` PARTITION(`p0`, `p1`) DISTRIBUTIONS"},
 		{"show table t1 partition (p0,p1) distributions where a=1", true, "SHOW TABLE `t1` PARTITION(`p0`, `p1`) DISTRIBUTIONS WHERE `a`=1"},
 
+		// for distribute table
+		{"distribute table t1", false, ""},
+		{"distribute table t1 partition(p0)", false, ""},
+		{"distribute table t1 partition(p0,p1)", false, ""},
+		{"distribute table t1 partition(p0,p1) engine = tikv", false, ""},
+		{"distribute table t1 rule = leader engine = tikv", true, "DISTRIBUTE TABLE `t1` RULE = `leader` ENGINE = `tikv`"},
```

Comment:
I think we'd better combine all the options:
- partition vs non-partition table
- rule = leader vs peer
- engine = tikv vs tiflash

Reply:
the values of rule and engine are just literal, so all the combinations should pass. 
We can check the args in the process of the execution.